File: lib/models/product.dart
class Product {
  final int id;
  final String name, category, brand, description, imageAsset, unit, deliveryType;
  final double mrp, sellingPrice;
  final int stock;
  final bool available;
  const Product({required this.id, required this.name, required this.category, required this.brand, required this.description, required this.imageAsset, required this.unit, required this.mrp, required this.sellingPrice, required this.stock, required this.available, required this.deliveryType});
  double get discount => mrp <= 0 ? 0 : ((mrp-sellingPrice)/mrp)*100;
  factory Product.fromJson(Map<String,dynamic> j) => Product(id: j['id'] ?? 0, name: j['name'] ?? '', category: j['categoryName'] ?? j['category'] ?? '', brand: j['brand'] ?? '', description: j['description'] ?? '', imageAsset: j['imageAsset'] ?? '', unit: j['unit'] ?? '1 unit', mrp: (j['mrp'] ?? 0).toDouble(), sellingPrice: (j['sellingPrice'] ?? 0).toDouble(), stock: j['stockQuantity'] ?? 0, available: j['available'] ?? true, deliveryType: j['deliveryType'] ?? 'SMALL');
}

File: backend/src/main/java/com/quickcart/entity/Product.java
package com.quickcart.entity;
import jakarta.persistence.*; import lombok.*;
@Entity @Table(name="products") @Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class Product { @Id @GeneratedValue(strategy=GenerationType.IDENTITY) private Long id; @Column(nullable=false) private String name; @Column(length=2000) private String description; private String brand; @ManyToOne(fetch=FetchType.EAGER,optional=false) private Category category; private double mrp; private double sellingPrice; private double discountPercentage; private String unit; private String weight; private String size; private boolean available=true; private int stockQuantity; private String deliveryType; private String imageAsset; }
